### ISSUE TITLE
False positive missing keys when using Trans from 'react-i18next'

### ISSUE THREAD

**Issue by filipjakov** [lingualdev/i18n-check#22]
Getting false positive missing keys when using Trans from 'react-i18next'.

Example:

```
<Text textStyle="body2">
	<TypeSafeTrans
		i18nKey="accounts:account.tabs.positions.table.footer.closeAll.modal.subtitle"
		components={[
			<Fragment key="0">
				<br />
				<br />
			</Fragment>,
		]}
	/>
</Text>
```

or

```
<Text textStyle="body2">
	<TypeSafeTrans
		i18nKey="account.tabs.positions.table.footer.closeAll.modal.subtitle"
                ns="accounts" // <- separated namespace
		components={[
			<Fragment key="0">
				<br />
				<br />
			</Fragment>,
		]}
	/>
</Text>
```

and running: `i18n-check -s en --locales public/locales -f i18next -u src -f i18next` gets `Found unused keys: account.tabs.positions.table.footer.closeAll.modal.subtitle`

Am I missing some configuration or is this not yet supported?

**Comment by busypeoples** [comment]
Thanks for the letting us know about the issue @filipjakov. It should technically work, as we our using the actual i18next-parser under the hood. Let me try to use your example and replicate the issue and then try to find a solution. Can we use a regular `Trans` component to replicate the issue, or is `TypeSafeTrans` doing something specific, that we need to consider?

**Comment by busypeoples** [comment]
@filipjakov ok, will we be able to fix it. Will try to introduce a way to tell what the component function names are, if the `Trans` component is not used directly. We might probably add the option to add a configuration, as it might be easier to configure, than using command line options. Will let you know, when we have the solution rolled out, so you can test again.

**Comment by filipjakov** [comment]
Sorry, my bad. It's a regular`Trans` component, just re-exported internally in the project.

Thanks 🙏

**Comment by busypeoples** [comment]
The fix will be in the next release.

**Comment by busypeoples** [comment]
It's possible to define the component names to parse for in i18next via the [--parser-component-functions](https://github.com/lingualdev/i18n-check?tab=readme-ov-file#--parser-component-functions) option in the latest release (v0.5.1).

Pass the additional component names like so:

```ts
yarn i18n:check --locales translations/i18NextMessageExamples -s en-US -f i18next
-u src --parser-component-functions WrappedTransComponent


yarn i18n:check --locales translations/i18NextMessageExamples -s en-US -f i18next
-u src --parser-component-functions WrappedTransComponent AnotherWrappedTransComponent
```

**Comment by filipjakov** [comment]
Tried, seems it works partially

for example, this is not a false positive anymore:

```tsx
<TypeSafeTrans
	i18nKey="user.id"
	ns="users"
	components={[<Text key="0" as="span" />, <Text key="1" as="span" isTruncated />]}
	values={{ id: data.id }}
/>
```

but this still is:

```tsx
<TypeSafeTrans
	i18nKey="account.tabs.positions.table.footer.closeAll.modal.subtitle"
	ns="accounts"
	components={[
		<Fragment key="0">
			<br />
			<br />
		</Fragment>,
	]}
/>
```

not sure what difference could be? do you need more context?

**Comment by busypeoples** [comment]
That's an interesting scenario. Can't really see any difference between the two components. Will copy the two above components and try to see if I can replicate the issue.

**Comment by filipjakov** [comment]
@busypeoples thank you, everything works as expected on 0.5.2

Just FYI, maybe not an issue, but when having no unused keys, the following state appears after running the cli:

![Image](https://github.com/user-attachments/assets/7a3815c9-381d-4583-a2bf-026e489c8a57)

**Comment by busypeoples** [comment]
Thanks for the info @filipjakov  Will check what is happening here and create a separate issue for that.
